public/app/plugins/panel/market-trend/utils.ts

```diff
@@ -0,0 +1,167 @@
+import { MarketTrendMode, ColorStrategy, PriceStyle } from './types';
+import uPlot from 'uplot';
+import { colorManipulator } from '@grafana/data';
+
+const { alpha } = colorManipulator;
+
+export function drawMarkers(opts) {
+  let { mode, priceStyle, fields, colorStrategy, upColor, downColor, flatColor, volumeAlpha, flatAsUp = true } = opts;
```

Comment:
@leeoniya -- this is the function that is the biggest blocker for types, can you take a stab at it?  The types can be defined inline and just includes types so the rest of us mortals have a chance

```suggestion
export function drawMarkers({ 
  mode: ???, 
  priceStyle: ???, 
  fields: ???, 
  colorStrategy, 
  upColor, 
  downColor, 
  flatColor, 
  volumeAlpha, 
  flatAsUp = true } ) {
```